File: backend/routes/rule.go
package routes

import (
	"app/controllers/rule"
	"app/middlewares"

	"github.com/gin-gonic/gin"
)

func InitRule(r *gin.RouterGroup, e *gin.Engine) {
	const baseUrl = "/:id/rules"

	r.GET(baseUrl + "/", middlewares.CheckWorkspace, rule.Index)
	r.POST(baseUrl + "/", middlewares.CheckWorkspace, rule.Create)
	r.PUT(baseUrl + "/:ruleId", middlewares.CheckWorkspace, rule.Update)
	r.DELETE(baseUrl + "/:ruleId", middlewares.CheckWorkspace, rule.Delete)
}

File: frontend/src/api/rule.ts
import http from './_http';

export default {
  index (workspaceId: number) {
    return http.get(`/workspaces/${workspaceId}/rules/`);
  },
  create (workspaceId: number, data: any) {
    return http.post(`/workspaces/${workspaceId}/rules/`, data);
  },
  update (workspaceId: number, id: number, data: any) {
    return http.put(`/workspaces/${workspaceId}/rules/${id}`, data);
  },
  delete (workspaceId: number, id: number) {
    return http.delete(`/workspaces/${workspaceId}/rules/${id}`);
  },
}